drivers/counter/counter_gecko_rtcc.c

```diff
@@ -219,12 +223,68 @@ static uint32_t counter_gecko_get_pending_int(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PERSISTENT_GECKO_RTCC
+/*! /brief Set the posix time to the RTCC value.
+ *
+ *  This is ONLY done when we are preserving the RTCC operations in EM4. If the
+ *  RTCC power has been removed, then the RTCC value is not saved, but both
+ *  values will be the same anyway.
+ */
+static int update_posix(void)
+{
+	struct timespec tp;
+
+	clock_gettime(CLOCK_REALTIME, &tp);
+	tp.tv_sec = RTCC_CounterGet();
+
+	return clock_settime(CLOCK_REALTIME, &tp);
+}
+
+/*! \brief Pin bootloader definitions.
+ *
+ * The UART Bootloader (e.g. First Stage Bootloader) also uses the RTCC so it must
+ * be disabled if we are preserving the time in the RTCC.
+ *
+ * Here is a list of the devices supported by the UART Bootloader:
+ * See: https://www.silabs.com/documents/public/application-notes/an0003-efm32-uart-bootloader.pdf
+ * Note that the CONFIG symbols are artificial when they are terminated with a $.
+ */
+#if    (/* EFM32 Series 0: */ \
+	defined(CONFIG_SOC_SERIES_EFM32G$)	/*!< EFM32 Gecko */ ||	\
+	defined(CONFIG_SOC_SERIES_EFM32GG$)	/*!< EFM32 Giant Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EFM32WG)	/*!< EFM32 Wonder Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EFM32LG$)	/*!< EFM32 Leopard Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EFM32TG$)	/*!< EFM32 Tiny Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EFM32ZG$)	/*!< EFM32 Zero Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EFM32HG$)	/*!< EFM32 Happy Gecko */ || \
+	/* EZR32 Series 0: */ \
+	defined(CONFIG_SOC_SERIES_EZR32WG$)	/*!< EZR32 Wonder Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EZR32LG$)	/*!< EZR32 Leopard Gecko */ || \
+	defined(CONFIG_SOC_SERIES_EZR32HG$)	/*!< EZR32 Happy Gecko */ || \
+	/* EFM32 Series 1: */ \
+	defined(CONFIG_SOC_SERIES_EFM32PG1$)	/*!< EFM32 Pearl Gecko (Rev C onwards) */ || \
+	defined(CONFIG_SOC_SERIES_EFM32PG12B)	/*!< EFM32 Pearl Gecko (Rev C onwards) */ || \
+	defined(CONFIG_SOC_SERIES_EFM32JG1$)	/*!< EFM32 Jade Gecko (Rev C onwards) */ || \
+	defined(CONFIG_SOC_SERIES_EFM32JG12$)	/*!< EFM32 Jade Gecko (Rev C onwards) */ || \
+	defined(CONFIG_SOC_SERIES_EFM32GG11$)	/*!< EFM32 Giant Gecko GG11 */ || \
+	defined(CONFIG_SOC_SERIES_EFM32GG12$)	/*!< EFM32 Giant Gecko GG12 */ || \
+	defined(CONFIG_SOC_SERIES_EFM32TG11$)	/*!< EFM32 Tiny Gecko 11 */)
+#define LB_CLW0           (*((volatile uint32_t *)(LOCKBITS_BASE)+122))
+#define LB_CLW0_PINBOOTLOADER    (1 << 1)
+#endif /* EFM32, EZR32 SOC Series */
+#endif /* PERSISTENT_GECKO_RTCC */
```

Comment:
IMO this belongs in the SoC specific startup code, not in the driver itself.

Reply:
In zephyrproject/zephyr/soc/arm/silabs_exx32/common/soc.c?